### PR TITLE
Merging CDC2021 - 2022 from JC

### PR DESCRIPTION
**Pull request by JulienCalbert**

### CODE REVIEW

File: src/Dionysos.jl [hunk 1]
```diff
@@ -6,6 +6,6 @@ include("system/system.jl")
 include("search/search.jl")
 include("symbolic/symbolic.jl")
 include("control/control.jl")
-include("problem/problem.jl")
+# include("problem/problem.jl")
```
Comment (blegat) [inline src/Dionysos.jl:9]: Why is this commented?

**Comment by blegat** [comment]
Can `src/CDC2021/tests` be moved to `test/CDC2021`?
The tests are failing, could you take a look?

File: Project.toml [hunk 2]
```diff
@@ -4,24 +4,38 @@ authors = ["Raphael Jungers <raphael.jungers@uclouvain.be>", "Antoine Aspeel <an
...
 HybridSystems = "2207ec0c-686c-5054-b4d2-543502888820"
+InteractiveUtils = "b77e0a4c-d291-57a0-90e8-8db25a27a240"
+IntervalArithmetic = "d1acc4aa-44c8-5952-acd4-ba5d80a2a253"
 JuMP = "4076af6c-e467-56ae-b986-b466b2749572"
 LazySets = "b4f0291d-fe17-52bc-9479-3d1a343d9043"
-LightXML = "9c8b4983-aa76-5018-a973-4c85ecc9e179"
+LightGraphs = "093fc24a-ae57-5d10-9952-331d41423f4d"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MathematicalSystems = "d14a8603-c872-5ed3-9ece-53e0e82e39da"
+MeshCat = "283c5d60-a78f-5afe-a0af-af636b173e11"
```
Comment (blegat) [inline Project.toml:25]: Can you move dependencies of the examples that are not dependencies of Dionysos to `examples/Project.toml`?

File: examples/example_nested_domain.jl [hunk 3]
```diff
@@ -0,0 +1,39 @@
+
+module Test
+include("../Dionysos.jl")
```
Comment (egidioln) [inline examples/example_nested_domain.jl:3]:
```suggestion
using Dionysos
```

File: src/CDC2021/TODO.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+- add the idea of overlepping cells
+- for the moment the discretization step size is hard coded in the algotihtm 
+- improve the computation of the lower-bound
+- several heuristics for the branch-and-bound
+- add multi-threading
```
Comment (egidioln) [inline src/CDC2021/TODO.txt:5]: Maybe you don't wanna commit this file but rather add its contents to an issue

File: src/CDC2021/lazy_abstraction.jl [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+"""
+Module that aims to solve a reachability problem.
+For this, it builds an abstraction that is alternatingly simulated by our original system.
+This abstraction and the controller are built lazily based on a heuristic.
+"""
+#TO DO: -array vs dictionnary
+#       -eventually delete State struct
+
+include("search.jl")
+using .Search
```
Comment (egidioln) [inline src/CDC2021/lazy_abstraction.jl:11]:
```suggestion
using ..Search
```

Doesn't this work as the others? I wonder if this ``..`` are the best way to include it...

File: src/CDC2021/tests/test_articulated_vehicle.jl [hunk 6]
```diff
@@ -0,0 +1,395 @@
+include(joinpath("..", "..", "Abstraction", "abstraction.jl"))
+include("../general_domain.jl")
+include("../utils.jl")
+include("../alternating_simulation.jl")
+include("../partition.jl")
+include("../lazy_abstraction.jl")
+include("../branch_and_bound.jl")
+include("../optimal_control.jl")
```
Comment (egidioln) [inline src/CDC2021/tests/test_articulated_vehicle.jl:8]: I think these should be accessed through Dionysos package and not directly through the files like this...

File: src/CDC2021/tests/test_articulated_vehicle_2.jl [hunk 7]
```diff
@@ -0,0 +1,420 @@
+include(joinpath("..", "..", "Abstraction", "abstraction.jl"))
+include("../general_domain.jl")
+include("../utils.jl")
+include("../alternating_simulation.jl")
+include("../partition.jl")
+include("../lazy_abstraction.jl")
+include("../branch_and_bound.jl")
+include("../optimal_control.jl")
```
Comment (egidioln) [inline src/CDC2021/tests/test_articulated_vehicle_2.jl:8]: Same here

File: src/CDC2021/tests/test_lazy_periodicity.jl [hunk 8]
```diff
@@ -0,0 +1,212 @@
+include(joinpath("..", "..", "Abstraction", "abstraction.jl"))
+include("../general_domain.jl")
+include("../utils.jl")
+include("../lazy_abstraction.jl")
+include("../alternating_simulation.jl")
```
Comment (egidioln) [inline src/CDC2021/tests/test_lazy_periodicity.jl:5]: Same here 2

File: src/CDC2021/tests/test_pendulum.jl [hunk 9]
```diff
@@ -0,0 +1,600 @@
+include(joinpath("..", "..", "Abstraction", "abstraction.jl"))
+include("../general_domain.jl")
+include("../utils.jl")
+include("../alternating_simulation.jl")
+include("../partition.jl")
+include("../lazy_abstraction.jl")
+include("../branch_and_bound.jl")
+include("../optimal_control.jl")
```
Comment (egidioln) [inline src/CDC2021/tests/test_pendulum.jl:8]: same here 3

File: src/CDC2021/tests/test_simple_system.jl [hunk 10]
```diff
@@ -0,0 +1,209 @@
+include(joinpath("..", "..", "Abstraction", "abstraction.jl"))
+include("../general_domain.jl")
+include("../utils.jl")
+include("../alternating_simulation.jl")
+include("../partition.jl")
+include("../lazy_abstraction.jl")
+include("../branch_and_bound.jl")
+include("../optimal_control.jl")
```
Comment (egidioln) [inline src/CDC2021/tests/test_simple_system.jl:8]: same here 4

File: src/CDC2022/example_deformed_grid.jl [hunk 11]
```diff
@@ -0,0 +1,87 @@
+module Test
+include("../Dionysos.jl")
+using Dionysos, StaticArrays, Plots
```
Comment (egidioln) [inline src/CDC2022/example_deformed_grid.jl:3]: Why ``include`` and ``using Dionysos``?

File: src/CDC2022/example_nested_symbolic.jl [hunk 12]
```diff
@@ -0,0 +1,60 @@
+module Test
+include("../Dionysos.jl")
```
Comment (egidioln) [inline src/CDC2022/example_nested_symbolic.jl:2]: Include and using here too

File: src/CDC2022/system.jl [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+end
+
+function plot_limit_cycle(symmodel,sys)
+    l = 12
+    fig = plot(aspect_ratio = 1,legend = false,title="steady-state probability", xlim=(-l, l), ylim=(-l,l))
+    plot_steady_state!(symmodel,fact=0.0,color=:blue)
+    x0 = SVector(2.0,2.0)
+    plot_trajectory!(sys,x0,200)
+    display(fig)
+end
```
Comment (egidioln) [inline src/CDC2022/system.jl:167]: Maybe this file should have a more self-explanatory name as ``van_der_pol_oscillator.jl``?

File: src/domain/Project.toml [hunk 14]
```diff
@@ -1,2 +0,0 @@
-[deps]
```
Comment (egidioln) [inline src/domain/Project.toml:1]: Is it normal that there is a ``Project.toml`` here?

File: test/control/test_controllerreach.jl [hunk 15]
```diff
@@ -78,7 +75,7 @@ if VERSION >= v"1.5"
         @allocated CO._compute_controller_reach!(contr, autom, initset, targetset, num_targets_unreachable, current_targets, next_targets)
     end
     f(symmodel.autom, initlist, targetlist)
-    @test f(symmodel.autom, initlist, targetlist) == 3224896 # check this
+    @test f(symmodel.autom, initlist, targetlist) == 1635072 # check this
```
Comment (egidioln) [inline test/control/test_controllerreach.jl:78]: Why did this test changed, would you know?

**Comment by blegat** [comment]
It seems the branch has conflict. I would recomment squashing all commits of the branch into one commit and then rebasing on master.

**Comment by egidioln** [comment]
> It seems the branch has conflict. I would recomment squashing all commits of the branch into one commit and then rebasing on master.

I think that's because we modified Project.toml in the github interface yesterday.. If we revert it the conflict should be gone

**Comment by blegat** [comment]
I reverted it, the conflict is now gone

File: Acrobot.urdf [hunk 16]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0"?>
```
Comment (blegat) [inline Acrobot.urdf:1]: Remove this file

File: doublependulum.urdf [hunk 17]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0"?>
```
Comment (blegat) [inline doublependulum.urdf:1]: Remove this file if it is not needed. Otherwise, put it in the `examples` folder

File: src/domain/grid.jl [hunk 18]
```diff
@@ -1,5 +1,8 @@
-abstract type Grid{N,T} end
+using LazySets, Polyhedra
+using PyPlot, CDDLib
```
Comment (blegat) [inline src/domain/grid.jl:2]: Where are PyPlot and CDDLib needed here ?

**Comment by egidioln** [comment]
Hey tests are not passing (PyPlot-related problem) and CI doc job is running forever... Any clue what's happening? @JulienCalbert @blegat

**Comment by blegat** [comment]
Fixed in https://github.com/dionysos-dev/Dionysos.jl/pull/130

**Pull request merged**